libcaf_core/caf/policy/work_stealing.hpp

```diff
@@ -85,7 +85,8 @@ class work_stealing : public unprofiled {
       return nullptr;
     }
     // roll the dice to pick a victim other than ourselves
-    size_t victim = d(self).rengine() % (p->num_workers() - 1);
+    std::uniform_int_distribution<size_t> uniform(0, p->num_workers() - 2);
```

Comment:
> N workers excluding self are N - 1 workers. 

If the idea is to exclude self, then I don't think this is statistically correct, because the two distributions `Unif(0, N-1)` and `Unif(0, N-2)` have two different probabilities for each victim. Using `Unif(0, N-2)` would be correct if we can ensure that the current worker is always `N-1`. But I don't see the code doing that. Instead, there is a candidate check that checks if the selection hit ourselves:

```cpp
    if (victim == self->id())
      victim = p->num_workers() - 1;
```

What we want is a random variable that describes all indices *but* the current one. Since there's not straight-forward distribution for that, we use a candidate check. In fact, I think the candidate check should not pick always the last element, but something dynamic. Let's hold that thought. Ultimately, we want to sample from the remaining workers, but I think we have to do it with probability `1/N` instead of `1/(N-1)`, because the currently executing worker cannot be assumed to be `N`.